src/figuresClasses.ts

```diff
@@ -1,11 +1,88 @@
-export interface Figure {}
+function roundDown(number: number, decimals: number): number {
+  const decims = decimals || 0;
 
-export class Triangle implements Figure {}
+  return Math.floor(number * Math.pow(10, decims)) / Math.pow(10, decims);
+}
+
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to declare `shape` as a class field separately since you can use the value 'triangle' directly where needed or you can declare it as readonly property inside the class.